### PR TITLE
Review EE code in subquery branch

### PR DESCRIPTION
**Pull request by wolffcm**
Not many changes here, I just added some comments to clarify my understanding of the code.

### CODE REVIEW

File: src/ee/plannodes/plannodefragment.cpp [hunk 1]
```diff
@@ -89,8 +89,15 @@ void PlanNodeFragment::constructTree(AbstractPlanNode* node)
 
 PlanNodeFragment::~PlanNodeFragment()
 {
-    for (PlanNodeMapIterator mapIt = m_stmtExecutionListMap.begin(); mapIt != m_stmtExecutionListMap.end();) {
+    // The need to delete this could be avoided just by storing
+    // an actual instance (instead of a pointer) in the map?
+    //
+    // Also, why is the call to erase needed below?
```
Comment (jxchinaren) [inline src/ee/plannodes/plannodefragment.cpp]: @wolffcm Do you plan to create a new ticket to address this issue or just leave with it?

File: src/ee/execution/VoltDBEngine.cpp [hunk 2]
```diff
@@ -300,9 +303,15 @@ class ExecutorVector {
...
 {
+    // We could avoid having call delete here if we stored an instance in the
+    // map, rather than a pointer to it?
+    //
+    // Also, why do we need to call erase here?
     map<int, vector<AbstractExecutor*>* >::iterator it = m_subplanExecListMap.begin();
     while (it != m_subplanExecListMap.end()) {
         vector<AbstractExecutor*>* executorList = it->second;
+        // Note: we need to increment the iterator here
+        // to avoid invalidating it.
```
Comment (jxchinaren) [inline src/ee/execution/VoltDBEngine.cpp]: it++, means to use it first and increment later. So it's probably fine here, right?

**Comment by jxchinaren** [comment]
Looks good to me.

File: src/ee/plannodes/plannodefragment.cpp (hunk 1 above)
Comment (wolffcm) [inline src/ee/plannodes/plannodefragment.cpp]: I'll file a (very low priority) ticket for the minor nits that I found during review.

File: src/ee/execution/VoltDBEngine.cpp (hunk 2 above)
Comment (wolffcm) [inline src/ee/execution/VoltDBEngine.cpp]: Maybe this comment is confusing.  If one just does `erase(it)`, then `it` is invalidated, and attempting to increment an invalid iterator leads to crashes, etc.  Doing `erase(it++)` means that the iterator is incremented before `erase` completes, so the iterator is still valid on the next iteration of the loop.

I had originally tried to restructure this loop to a more familiar idiom, but ran into this problem, so I just added the comment for clarity.

Comment (jxchinaren) [inline src/ee/execution/VoltDBEngine.cpp]: I see what you mean. Now it seems to be a serious problem that can crash. Right?

Comment (wolffcm) [inline src/ee/execution/VoltDBEngine.cpp]: Now the loop works as intended, as it did before.  I just added the comment to clarify the subtleties.

**Comment by wolffcm** [comment]
Merging this now, will address comments on a subsequent commit, once valgrind is passing.

**Pull request merged**